recordedit/form.controller.js

```diff
@@ -252,16 +252,16 @@
                     fn = "update", fnScope = $rootScope.reference, args = [$rootScope.tuples];
                 }
 
-                fnScope[fn].apply(fnScope, args).then(function success(page) {
+                fnScope[fn].apply(fnScope, args).then(function success(result) {
 
+                    var page = result.successful;
+                    var failedPage = result.failed;
+
+                    // the returned reference is contextualized and we don't need to contextualize it again
                     var resultsReference = page.reference;
 
                     if (isUpdate) {
-                        resultsReference = $rootScope.reference.contextualize.compact;
-                        // check if there is a window that opened the current one
-                        // make sure the update function is defined for that window
-                        // verify whether we still have a valid vaue to call that function with
-                        if (window.opener && window.opener.updated && context.queryParams.invalidate) {
+                        if (window.opener && window.opener.updated) {
```

Comment:
Looks like you overwrote this change that I made yesterday to address the `window.update` issue. These changes are from PR #1146.